### PR TITLE
[ci skip] Revamp Configuration docs page.

### PR DESCRIPTION
**Pull request by dafriend**
Revised configuration.rst as discussed in issue #2554

### CODE REVIEW

File: user_guide_src/source/general/configuration.rst [hunk 1]
```diff
@@ -1,125 +1,151 @@
...
 	// Get shared instance with config function
-	$config = config( 'Pager' );
+	$config = config('Pager');
 
 	// Access config class with namespace
 	$config = config( 'Config\\Pager' );
 
-	// Access settings as class properties
+- By using ``\Config\Services::class`` which will apply an instance of the class's config for you::
+
```
Comment (lonnieezell) [inline user_guide_src/source/general/configuration.rst]: Services are different than config files. Please remove the sections about services here.

File: user_guide_src/source/general/configuration.rst [hunk 2]
```diff
@@ -132,35 +158,53 @@ prefix, followed by a dot (.), and then the variable name itself::
...
+    Config\App.CSPEnabled = true
+
+
+.. note:: Both the namespace prefix and the property name are case-sensitive. They must exactly match the full namespace and property names as defined in the configuration class file.
 
-When you instantiate a configuration file, any namespaced environment variables
-are considered for merging into the configuration objects' properties.
+The same holds for a *short prefix*, which is a namespace using only the lowercase version of 
+the configuration class name. If the short prefix matches the class name, 
+the value will from **.env** replaces the configuration file value.
```
Comment (lonnieezell) [inline user_guide_src/source/general/configuration.rst]: I'm guessing `will` should be removed from this sentence?

Comment (dafriend) [reply user_guide_src/source/general/configuration.rst]: Done.

File: user_guide_src/source/general/configuration.rst (hunk 1 above)
Comment (dafriend) [reply user_guide_src/source/general/configuration.rst]: Yes, but when you use Services the associated config is (always?) instantiated and applied for you. That was the point I was trying to make. 
Not germane?

Comment (MGatner) [reply user_guide_src/source/general/configuration.rst]: I think it is convenient but I had to read it a few times to realize what it meant. I think it’s better left out of this part but could maybe have a life as a bonus “note” in Services?

**Comment by MGatner** [comment]
Very nice pass at this whole section, thanks for the work!

**Comment by dafriend** [comment]
Associated issue #2554 was addressed by this merge and should be closed

**Pull request merged**